pghoard/pghoard.py

```diff
@@ -609,6 +622,23 @@ def startup_walk_for_missed_files(self):
 
                 filetype = FileType.Timeline if wal.TIMELINE_RE.match(filename) else FileType.Wal
 
+                # verify if file was already compressed, otherwise the transfer agent will encounter
+                # duplicated UploadEvents. In case it was compressed, we should just add it to the deletion queue
+                base_compressed_file_path = (
+                    compressed_timeline_path if filetype is FileType.Timeline else compressed_xlog_path
+                )
+                if os.path.isfile(os.path.join(base_compressed_file_path, filename)):
+                    self.log.debug(
+                        "Uncompressed file %r is already compressed, adding to deletion queue.", full_path
+                    )
+                    if filetype is FileType.Timeline:
+                        os.unlink(full_path)
```

Comment:
For timelines I am just applying what we do in the compressor. Adding to the deletion queue might require changes on the compressor as well (in case we want the same logic there)